doc/tutorial/index.rst

```diff
@@ -184,6 +197,254 @@ as before, or leverage the convenience script as follows:
+root of the "table of contents tree" (or *toctree*).  Sphinx allows you to
+assemble a project from different files, which is helpful when the project
+grows.
+
+As an example, create a new file ``docs/source/usage.rst`` (next to
+``index.rst``) with these contents:
+
+.. code-block:: rest
+   :caption: docs/source/usage.rst
+
+   Usage
+   =====
+
+   Installation
+   ------------
+
+   To use Lumache, first install it using pip:
+
+   .. code-block:: console
+
+      (.venv) $ pip install lumache
+
+This new file contains two :ref:`section <rst-sections>` headers, normal
+paragraph text, and a :ref:`code-block <rst-code-block>` directive that renders
+a block of content as source code, with appropriate syntax highlighting
+(in this case, generic ``console`` text).
+
+The structure of the document is determined by the succession of heading
+styles, which means that, by using ``---`` for the "Installation" section
+after ``===`` for the "Usage" section, you have declared "Installation" to
+be a *subsection* of "Usage".
+
+To complete the process, add a ``toctree`` :ref:`directive <rst-directives>` at
+the end of ``index.rst`` including the document you just created, as follows:
+
+.. code-block:: rest
+   :caption: docs/source/index.rst
+
+   Contents
+   --------
+
+   .. toctree::
+
+      usage
+
+This step inserts that document in the root of the *toctree*, so now it belongs
+to the structure of your project, which so far looks like this:
+
+.. code-block:: text
+
+   index
+   └── usage
+
+If you export the documentation to HTML running ``make html``, you will see
+that the ``toctree`` gets rendered as a list of hyperlinks, and this allows you
+to navigate to the new page you just created.  Neat!
+
+.. warning::
+
+   Documents outside a *toctree* will result in ``WARNING: document isn't
+   included in any toctree`` messages during the build process, and will be
+   unreachable for users.
+
+Adding cross-references
+~~~~~~~~~~~~~~~~~~~~~~~
+
+One powerful feature of Sphinx is the ability to seamlessly add
+:ref:`cross-references <xref-syntax>` to specific parts of the documentation:
+a document, a section, a figure, a code object, etc.  This tutorial is full of
+them!
+
+To add a cross-reference, write this sentence right after the
+introduction paragraph in ``index.rst``:
+
+.. code-block:: rest
+   :caption: docs/source/index.rst
+
+   Check out the :doc:`usage` section for further information.
+
+The :rst:role:`doc` role you used automatically references a specific document
+in the project, in this case the ``usage.rst`` you created earlier.
+
+Alternatively, you can also add a cross-reference to an arbitrary part of the
+project. For that, you need to use the :rst:role:`ref` role, and add an
+explicit *label* that can act as a target.
+
+For example, to reference the "Installation" subsection, add a label right
+before the heading, as follows:
+
+.. code-block:: rest
```

Comment:
We usually uses `rst` as a highlight language for reStructuredText.

```
tkomiya@Alrescha> grep -r code-block:: doc | grep ':: rst'
doc/man/sphinx-autogen.rst:.. code-block:: rst
doc/development/theming.rst:.. code-block:: rst
doc/development/tutorials/todo.rst:.. code-block:: rst
doc/development/tutorials/todo.rst:.. code-block:: rst
doc/development/tutorials/todo.rst:.. code-block:: rst
doc/development/tutorials/helloworld.rst:.. code-block:: rst
doc/development/tutorials/helloworld.rst:.. code-block:: rst
doc/development/tutorials/recipe.rst:.. code-block:: rst
doc/development/tutorials/recipe.rst:.. code-block:: rst
doc/development/tutorials/autodoc_ext.rst:.. code-block:: rst
doc/usage/configuration.rst:   .. code-block:: rst
doc/usage/extensions/doctest.rst:.. code-block:: rst
doc/faq.rst:  .. code-block:: rst
```

I noticed some code-blocks uses "rest"...
```
tkomiya@Alrescha> grep -r code-block:: doc | grep ':: rest'
doc/tutorial/index.rst:.. code-block:: rest
doc/tutorial/index.rst:.. code-block:: rest
doc/usage/quickstart.rst:.. code-block:: rest
doc/usage/quickstart.rst:.. code-block:: rest
doc/usage/quickstart.rst:.. code-block:: restructuredtext
doc/usage/quickstart.rst:.. code-block:: restructuredtext
```

Reply:
Note: Both `rest` and `rst` are supported.
https://pygments.org/docs/lexers/#pygments.lexers.markup.RstLexer

Reply:
Fair enough, I will convert everything to `rst` for consistency.